### DISCUSSION TITLE
How to create an org-capture template with a specified subdirectory?

### DISCUSSION THREAD

**Discussion by jtpavlock** [protesilaos/denote#448] [Q&A]
How can I define an org-capture template with a specified subdirectory?

For example, I'd like to create a "reference" org-capture template that automatically selects the "reference" subdirectory without prompting me for it.

**Comment by protesilaos** [comment]
> From: jtpavlock ***@***.***>
> Date: Sun, 22 Sep 2024 15:23:31 -0700
>
> How can I define an org-capture template with a specified subdirectory?
>
> For example, I'd like to create a "reference" org-capture template
> that automatically selects the "reference" subdirectory without
> prompting me for it.

Try this:

    (with-eval-after-load 'org-capture
      (add-to-list 'org-capture-templates
                   '("r" "New reference (with Denote)" plain
                     (file denote-last-path)
                     (function
                      (lambda ()
                        (let ((denote-use-directory "reference"))
                          (denote-org-capture))))
                     :no-save t
                     :immediate-finish nil
                     :kill-buffer t
                     :jump-to-captured t)))

-- 
Protesilaos Stavrou
https://protesilaos.com

**Reply by jtpavlock** [reply]
Thank you! This was exactly what I was looking for!

I tweaked your solution slightly as it didn't seem to recognize subdirectories, so I just had to give it the full directory path:

```elisp
  (with-eval-after-load 'org-capture
    (add-to-list 'org-capture-templates
                 '("r" "reference note" plain
                   (file denote-last-path)
                   (function
                    (lambda ()
                      (let ((denote-use-directory (concat denote-directory "reference")))
                        (denote-org-capture))))
                   :no-save t
                   :immediate-finish nil
                   :kill-buffer t
                   :jump-to-captured t)))
```

**Comment by protesilaos** [comment]
> From: jtpavlock ***@***.***>
> Date: Tue, 24 Sep 2024 08:15:02 -0700
>
> Thank you! This was exactly what I was looking for!

You are welcome!

> I tweaked your solution slightly as it didn't seem to recognize
> subdirectories, so I just had to give it the full directory path:

> [... 15 lines elided]

Very well!

-- 
Protesilaos Stavrou
https://protesilaos.com
